TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -1445,31 +1427,35 @@ currentIndex < (lastIndex+1) && (line[currentIndex + 2] == '"' || line[currentIn
                     int saveCurrentIndex = currentIndex;
                     Token numericLiteralToken = ScanNumericLiteral(startIndex);
 
-                    // 2. Then check to see if the next char would be valid inside a CobolWord
-                    bool nextCharIsACobolWordChar = (currentIndex <= lastIndex) && CobolChar.IsCobolWordChar(line[currentIndex]);
-                    if(nextCharIsACobolWordChar && line[currentIndex] == '-')
+
+                    // 2. a FloatingPointLiteral contains a character '.' or ',' which is invalid
+                    // in a UserDefinedWord or a keyword, so we can return this token directly.
+                    if (numericLiteralToken.TokenType == TokenType.FloatingPointLiteral)
```

Comment:
So this means that `ScanNumericLiteral` handles everything for FloatingPointLiteral ?

Reply:
Yes